pg_search/src/index/reader/index.rs

```diff
@@ -985,13 +1011,41 @@ impl SearchIndexReader {
                 OrderByInfo {
                     feature: OrderByFeature::Field(sort_field),
                     direction,
+                    semantic,
                 } => {
                     let field = self
                         .schema
                         .search_field(sort_field)
                         .expect("sort field should exist in index schema");
 
-                    match field.field_entry().field_type().value_type() {
+                    let original_value_type = field.field_entry().field_type().value_type();
+                    let mut resolved_type = original_value_type;
+                    if field.is_json() {
```

Comment:
This `if field.is_json() { .. } else if matches!(..) { .. }` block is duplicated in `search_top_n_in_segments`: please try to extract into a method.